src/store/thunks.ts

```diff
@@ -1101,7 +1101,27 @@ export const onUpdateRouter = (renderNode: RenderNode) => (
     updated = mutators.mergeNode(updated, renderNode);
   }
 
+  const routerWithCases = renderNode.node.router as any;
+  let updatedLocalization = definition.localization;
+
+  if (routerWithCases && routerWithCases.cases) {
+    routerWithCases.cases.forEach((caseItem: any) => {
+      if (caseItem.translations) {
+        Object.entries(caseItem.translations).forEach(([language, translation]: any) => {
+          if (!updatedLocalization[language]) {
+            updatedLocalization[language] = {};
+          }
+
+          updatedLocalization[language][caseItem.uuid] = {
+            arguments: translation.arguments
+          };
```

Comment:
Can you explain this part. Not sure what's happening here.

Reply:
this part is just updating the localization object with translations for the router